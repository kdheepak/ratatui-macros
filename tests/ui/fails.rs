use ratatui::prelude::*;
use ratatui_macros::constraints;

fn main() {
    constraints![,];

    // TODO: Make this compiler error pass
    let [a, b] = constraints![
      == 1/2,
      == 2,
    ];
    assert_eq!(a, Constraint::Ratio(1, 2));
    assert_eq!(b, Constraint::Length(2));

    let [a, b, c] = constraints![ == 1, == 10%, == 2; 4];
}
