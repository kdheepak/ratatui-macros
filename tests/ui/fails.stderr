error: No rules expected the token `,` while trying to match the end of the macro
Error:  --> tests/ui/fails.rs:5:5
  |
5 |     constraints![,];
  |     ^^^^^^^^^^^^^^^
  |
  = note: this error originates in the macro `$crate::constraints` which comes from the expansion of the macro `constraints` (in Nightly builds, run with -Z macro-backtrace for more info)

error: unexpected end of macro invocation
Error:   --> tests/ui/fails.rs:8:18
   |
8  |       let [a, b] = constraints![
   |  __________________^
9  | |       == 1/2,
10 | |       == 2,
11 | |     ];
   | |_____^ missing tokens in macro arguments
   |
note: while trying to match `==`
  --> src/layout.rs
   |
   |     (== $token:tt %) => {
   |      ^^
   = note: this error originates in the macro `$crate::constraints` which comes from the expansion of the macro `constraints` (in Nightly builds, run with -Z macro-backtrace for more info)

error: no rules expected the token `;`
Error:   --> tests/ui/fails.rs:15:53
   |
15 |     let [a, b, c] = constraints![ == 1, == 10%, == 2; 4];
   |                                                     ^ no rules expected this token in macro call
   |
note: while trying to match `%`
  --> src/layout.rs
   |
   |     (== $token:tt %) => {
   |                   ^

error: first parameter must be a formatting specifier followed by a comma OR a `Style` followed by a semicolon
Error:   --> tests/ui/fails.rs:17:13
   |
17 |     let _ = span!(Modifier::BOLD, "hello world");
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `span` (in Nightly builds, run with -Z macro-backtrace for more info)

error: argument never used
Error:   --> tests/ui/fails.rs:19:28
   |
19 |     let _ = span!("hello", "hello world");
   |                   -------  ^^^^^^^^^^^^^ argument never used
   |                   |
   |                   formatting specifier missing

error[E0527]: pattern requires 2 elements but array has 3
Error:  --> tests/ui/fails.rs:8:9
  |
8 |     let [a, b] = constraints![
  |         ^^^^^^ expected 3 elements

error[E0277]: the trait bound `Style: From<&str>` is not satisfied
Error:   --> tests/ui/fails.rs:21:19
   |
21 |     let _ = span!("hello"; "hello world");
   |             ------^^^^^^^----------------
   |             |     |
   |             |     the trait `From<&str>` is not implemented for `Style`, which is required by `&str: Into<Style>`
   |             required by a bound introduced by this call
   |
   = help: the following other types implement trait `From<T>`:
             <Style as From<crossterm::style::content_style::ContentStyle>>
             <Style as From<Color>>
             <Style as From<Modifier>>
             <Style as From<(Color, Color)>>
             <Style as From<(Color, Color, Modifier)>>
             <Style as From<(Color, Color, Modifier, Modifier)>>
             <Style as From<(Color, Modifier)>>
             <Style as From<(Modifier, Modifier)>>
   = note: required for `&str` to implement `Into<Style>`
note: required by a bound in `Span::<'a>::styled`
  --> $CARGO/ratatui-0.26.2/src/text/span.rs
   |
   |     pub fn styled<T, S>(content: T, style: S) -> Self
   |            ------ required by a bound in this associated function
...
   |         S: Into<Style>,
   |            ^^^^^^^^^^^ required by this bound in `Span::<'a>::styled`
